src/main/java/emu/grasscutter/game/entity/GameEntity.java

```diff
@@ -53,7 +53,7 @@ public GameEntity(Scene scene) {
     public abstract void initAbilities();
 
     public int getEntityType() {
-        return this.getId() >> 24;
+        return EntityIdType.toEntityType(this.getId() >> 24);
```

Comment:
Wouldn't it be better to actually return the enum type, if you do the conversion? Otherwise something might expect the wrong value again.